### PR TITLE
[MM-21473] Review archiveTeamsCmdF function to return an error in case of a failure

### PR DESCRIPTION
**Pull request by TealWater**
#### Summary

Used `multiError.Error` to return non-zero error values to the root handler when `mmctl` subcommand fail.  This PR involves only changing the `archiveTeamsCmdF` function.
#### Ticket Link

Fixes: https://github.com/mattermost/mattermost/issues/21473
#### Release Note

```release-note
none
```

### CODE REVIEW

File: server/cmd/mmctl/commands/team.go [hunk 1]
```diff
@@ -191,15 +192,20 @@ func archiveTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error
...
+			errorString = *multierror.Append(&errorString, errors.New("Unable to archive team '"+team.Name+"' error: "+err.Error()))
 		} else {
 			printer.PrintT("Archived team '{{.Name}}'", team)
 		}
 	}
 
+	if errorString.Len() > 0 {
+		return &errorString
+	}
 	return nil
```
Comment (hanzei) [inline server/cmd/mmctl/commands/team.go:209]: `ErrorOrNil` makes this easier to read.

```suggestion
	return result.ErrorOrNil()
```

File: server/cmd/mmctl/commands/team.go [hunk 2]
```diff
@@ -180,6 +180,7 @@ func deleteTeam(c client.Client, team *model.Team) (*model.Response, error) {
 }
 
 func archiveTeamsCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	var errorString multierror.Error
```
Comment (hanzei) [inline server/cmd/mmctl/commands/team.go:183]: Let's stick to the naming schema for other methods.
```suggestion
	var result *multierror.Error
```

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution @TealWater

**Review by hanzei — APPROVED** [review]

**Comment by TealWater** [comment]
@hanzei 

One of the tests for `mmctl` failed.  Here is the output:

```
Error Trace:	/mattermost/server/cmd/mmctl/commands/team_e2e_test.go:366
        	            				/mattermost/server/cmd/mmctl/commands/mmctl_test.go:92
        	            				/go/pkg/mod/github.com/stretchr/testify@v1.8.4/suite/suite.go:112
        	Error:      	Expected nil, but got: *multierror.Error{Errors:[]error{(*errors.errorString)(0xc00421aa00)}, ErrorFormat:(multierror.ErrorFormatFunc)(nil)}
```

I don't understand why this is an issue when we used `return result.ErrorOrNil()` which should return `nil`

not `*multierror.Error{Errors:[]error{(*errors.errorString)(0xc00421aa00)}, ErrorFormat:(multierror.ErrorFormatFunc)(nil)}`

**Comment by hanzei** [comment]
@TealWater The tests need to be updated. You can see that searching for a team fails, and we log an error. (See https://github.com/TealWater/mattermost/blob/a6eae2577934216ef88ee0a14fbc5d9403c63e9c/server/cmd/mmctl/commands/team_e2e_test.go#L341-L345)

**Comment by TealWater** [comment]
@hanzei 

Hmm, just a thought....

Given the fact that the error being returned is a string:

```
Error:      	Expected nil, but got: *multierror.Error{Errors:[]error{(*errors.errorString)(0xc00421aa00)}, ErrorFormat:(multierror.ErrorFormatFunc)(nil)}
```

Can I delete https://github.com/TealWater/mattermost/blob/a6eae2577934216ef88ee0a14fbc5d9403c63e9c/server/cmd/mmctl/commands/team_e2e_test.go#L370?

Maybe even require the type to `error.string`?

**Comment by hanzei** [comment]
Comparing the actual error via `EqualError` is a fine approach. One simpler solution is to use the `Error` method and check that any error was returned instead of the the `Nil` check that exits today.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by BenCookie95 — APPROVED** [review]

**Pull request merged**